### PR TITLE
Set PIF's IPv6 Gateway when in DHCP/Autconf

### PR DESCRIPTION
**Pull request by benjamreis**
Read the default route for IPv6 and set it in the PIF

### CODE REVIEW

File: ocaml/xapi/helpers.ml [hunk 1]
```diff
@@ -331,13 +331,20 @@ let update_getty () =
...
+              ~value:(Unix.string_of_inet_addr addr)
+        | None ->
+            ()
+    ) ;
+    match Net.Interface.get_ipv6_gateway dbg bridge with
+    | Some addr ->
+        Db.PIF.set_ipv6_gateway ~__context ~self:pif
+          ~value:(Unix.string_of_inet_addr addr)
+    | None ->
+        ()
```
Comment (psafont) [inline ocaml/xapi/helpers.ml:347]: Shouldn't this be inside the `if ...` block as well?

Comment (benjamreis) [reply ocaml/xapi/helpers.ml:347]: Yes it should

**Review by psafont — APPROVED** [review]

**Comment by benjamreis** [comment]
Validated with IPv6 DHCP & Autoconf (Static was already working and still is :+1: )

**Review by lindig — APPROVED** [review]

File: ocaml/networkd/bin/network_server.ml [hunk 2]
```diff
@@ -445,6 +445,24 @@ module Interface = struct
       )
       ()
 
+  let get_ipv6_gateway dbg name =
+    Debug.with_thread_associated dbg
+      (fun () ->
+        let output = Ip.route_show ~version:Ip.V6 name in
+        try
+          let line =
```
Comment (lindig) [inline ocaml/networkd/bin/network_server.ml:453]: I think this kind of problem lends itself to this style of coding:
```
output 
|> Astring.String.cuts ~empty:false ~sep:"\n"
|> List.find (fun s -> Astring.String.is_prefix ~affix:"default via" s)
```

**Pull request merged**